components/post_view/post/post.jsx

```diff
@@ -265,6 +265,7 @@ export default class Post extends React.PureComponent {
                 className={this.getClassName(post, isSystemMessage, fromWebhook, fromAutoResponder)}
                 onMouseOver={this.setHover}
                 onMouseLeave={this.unsetHover}
+                onTouchStart={this.setHover}
```

Comment:
Is the menu dismissed if you tap a post and then tap somewhere else? Is a mouse leave event fired in that case?

Reply:
Nope. it does not trigger an event but this works because css:hover class is removed once you tap another post. 

I am not sure if I should add any complexity to this code but i could add onTouchEnd event with a timeout to trigger the event so it autocloses after a certain time.

Any thoughts?

Reply:
Oh, I see. I don't think a timeout is an answer since we don't want it to hide without user interaction.

@jwilander You were the one who implemented the fix to not have the post menu render when the post isn't hovered, so you should probably weigh in here.

To summarize the problem, those changes made it so that the menu doesn't render on a tablet when it's in landscape and the web app shows in "desktop view". Because we don't have a reliable way to detect that you're on a tablet, Sudheer has added an `onTouchStart` to mark the post as hovered so that the menu shows up when the user taps on the post. When the user taps on another post, the menu is hidden by the existing CSS, but the hovered state isn't cleared, so the menu is still being rendered by React in the background. There's no other event that we can use to detect that the user tapped somewhere else (`onTouchEnd` is fired when the user lifts their finger), so there's no way to clear that state.

Do you think this behaviour is okay? It's closer to the behaviour from before [your performance PR changed this](https://github.com/mattermost/mattermost-webapp/pull/508), but it still won't be rendering them on first load. It's also an edge case since no one has reported that the post menus don't work on tablets since those changes went into the February release :p

Reply:
@hmhealey Summarised really well. Thanks :)

I have on thing to add. 
This touch start will be triggered only for tablets and so the performance concern would only be for tablets and on run-time. which I think should not amount to much as we don't re-render most of posts after initial channel load.

Unrelated to the PR: I think it might be better not to remove that element and hiding it as it will be faster to show it when needed again.

Reply:
Yeah, the performance concern is definitely minimal since, as I said, few enough people use this on a tablet that this bug wasn't reported for probably 8 months, but I wanted to double check.

Also unrelated: I was thinking about hiding vs not rendering elements like this earlier. It might be interesting to experiment with React's new Portal system which allows a component to render outside of its normal location. For dropdowns like this, we could see if that allows us to only render a single one of these menus offscreen and then attach it to a post whenever it's needed. Modals could work similarly.

Reply:
I thought about emoji picker working as a single entity in the page and position it based on the need. 

If react portal helps with this then it would really good change for emoji picker as well.

Reply:
No, each place it's used has an instance of the EmojiPickerOverlay component, but that only mounts the actual EmojiPicker when necessary so there's only a small bit of overhead by having multiple copies of it

Reply:
I'm good with it from a performance and implementation perspective 👍